### PR TITLE
[FLINK-2869] [tests] Apply JMH on IOManagerPerformanceBenchmark class.

### PR DESCRIPTION
**Pull request by gallenvara**
JMH is a Java harness for building, running, and analysing nano/micro/milli/macro benchmarks.Use JMH to replace the old micro benchmarks method in order to get much more accurate results.Modify the `IOManagerPerformanceBenchmark` class and move it to `flink-benchmark` module.

### CODE REVIEW

**Comment by gallenvara** [comment]
Part of the JMH benchmark result : 
jmhiomanagerbenchmark

File: flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+			// close if possible
+			if (daos != null) {
+				daos.close();
+			}
+		}
+		return tempFile;
+	}
+
+	@SuppressWarnings("resource")
+	private File creatSpeedTestNIOTempFile(int bufferSize, boolean direct) throws IOException
```
Comment (fhueske) [inline flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java]: should be creat**e**SpeedTestNIOTempFile

File: flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+			if (writer != null) {
+				writer.closeAndDelete();
+			}
+		}
+	}
+
+	private void testChannelReadWithSegments(int numSegments) throws Exception
+	{
+		final List<MemorySegment> memory = this.memManager.allocatePages(memoryOwner, numSegments);
+		//final FileIOChannel.ID channel = this.ioManager.createChannel();
```
Comment (fhueske) [inline flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java]: remove?

File: flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+	
+	private static File speedTestNIOTempFile4;
+
+
+	@Setup
+	public void startup() throws Exception {
+		memManager = new MemoryManager(MEMORY_SIZE, 1);
+		ioManager = new IOManagerAsync();
+		this.testChannelWriteWithSegments(numSegment);
+		ioManagerTempFile1 = this.createReadTempFile(segmentSizesAligned);
```
Comment (fhueske) [inline flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java]: are these files deleted after the benchmark completed?

Comment (gallenvara) [inline flink-benchmark/src/test/java/org/apache/flink/benchmark/runtime/io/disk/iomanager/IOManagerPerformanceBenchmark.java]: It's my fault that I forget to delete them.
Now i have add these operations to the `@TearDown` and submitted a new commit.

**Comment by fhueske** [comment]
Thanks for the update.
Looks good to merge.

**Comment by fhueske** [comment]
I will merge this PR.

**Pull request closed without merging**